dask/dataframe/backends.py

```diff
@@ -45,6 +46,50 @@
 from dask.sizeof import SimpleSizeof, sizeof
 from dask.utils import is_arraylike, is_series_like, typename
 
+
+class DataFrameBackendEntrypoint(DaskBackendEntrypoint):
```

Comment:
Unlike the array-API standard, I am unaware of anything mature(ish) for dataframes. Nevertheless, it feels unfortunate to me to have an abstract base class that doesn't put *any* constraints on the function signatures

Reply:
We can certainly specify more-specific kargs/kwargs in this class. However, it is not obvious to me if/how we can enforce that the subclass is in compliance.  As the class structure is currently defined, the subclass is free to use completely different args/kwargs that whatever we specify here. I suppose this could be fine- Do you have thoughts?

Reply:
> However, it is not obvious to me if/how we can enforce that the subclass is in compliance. 

In the absence of language support for enforcing API contracts / child-class method consistency, this is a job for linters/type checkers in my book. I believe that both mypy and pyright can do this. So it would be the responsibility of the implementing project and whatever code quality tools are being used.

> As the class structure is currently defined, the subclass is free to use completely different args/kwargs that whatever we specify here. I suppose this could be fine- Do you have thoughts?

My preference would be to try to write down *some* sort of base API set of signatures that implementations would be expected to match. Otherwise the dispatch system would really only be saying that different backends should have these methods defined, but they wouldn't be very swap-able. As a user, I would want to be able to change different backends with minimal code changes, and writing down a contract helps with that goal.

Reply:
Sorry for the delay here @ian-r-rose !  For now, I felt that it made the most sense to specify the required positional arguments, and to group all *optional* key-word arguments as `kwargs`. My reasoning is that it seems a bit difficult to find a clear distinction between **critical** and **nice-to-have** key-word arguments.  Does this seem reasonable for the inital (still-experimental) design?

Reply:
> If we use inheritance-based subtyping then type-checkers _will not_ complain about mismatching signatures:

This is true with mypy, but pyright has a nice option in `reportIncompatibleMethodOverride` which flags this snippet. I wish that mypy did the same.

> But if the abstract interface accepts arbitrary keyword arguments, then the implementing concrete classes must also accept them (so the signature is not fully self-documenting in terms of the valid keyword arguments). But at least we can specify additional named keyword arguments:

Yeah, it's unfortunate, but I'm not sure I see a way around this if we want to support different backends which might have different optional configurations. To me, being relatively strict about positional-only arguments, expected keyword arguments with their types, and then `**kwargs` is a reasonable compromise.